src/Services/CodeTypesService.php

```diff
@@ -429,4 +430,43 @@ public function parseCodesIntoCodeableConcepts($codes)
         }
         return $codeableConcepts;
     }
+
+    /**
+     * Return listing of pertinent and active code types.
+     *
+     * Function will return listing (ct_key) of pertinent
+     * active code types, such as diagnosis codes or procedure
+     * codes in a chosen format. Supported returned formats include
+     * as 1) an array and as 2) a comma-separated lists that has been
+     * process by urlencode() in order to place into URL  address safely.
+     *
+     * @param  string       $category       category of code types('diagnosis', 'procedure', 'clinical_term', 'active' or 'medical_problem')
+     * @param  string       $return_format  format or returned code types ('array' or 'csv')
+     * @return string|array
+     */
+    public function collectCodeTypes($category, $return_format = "array"): string|array
+    {
+        global $code_types;
+
+        // could turn this into an enum later if desired
+        if (!in_array($return_format, ['array','csv'])) {
+            throw new InvalidArgumentException("Unsupported return format: $return_format");
+        }
+
+        $code_remap = [
+            'active' => 'active',
+            'clinical_term' => 'term',
+            'diagnosis' => 'diag',
+            'drug' => 'drug',
+            'medical_problem' => 'problem',
+            'procedure' => 'proc',
+        ];
+        $cat_code = $code_remap[$category] ?? null;
```

Comment:
Oops, there may be a bug here in that I didn't think through what should happen if `$cat_code` ends up `null`. Kinda feels like that's an error condition.

Reply:
I'll have it throw an exception if the category is invalid.  The previous behavior was to just return an empty array, but I think we want people to know they've provided an invalid value.